lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -56,10 +56,9 @@ def script
             # use filter instead of exclude so missing patterns dont' throw errors
             echo "rsync -av --filter \"- CVS/\" --filter \"- .svn/\" --filter \"- .git/\" --filter \"- .hg/\" --filter \"- Headers\" --filter \"- PrivateHeaders\" --filter \"- Modules\" ${source} ${destination}"
             rsync -av --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "${source}" "${destination}"
+
             # Resign the code if required by the build settings to avoid unstable apps
-            if [ "${CODE_SIGNING_REQUIRED}" == "YES" ]; then
-                code_sign "${destination}/$1"
-            fi
+            code_sign_if_enabled "${destination}/$(basename $1)"
```

Comment:
You're right about quoting, in case the framework contains spaces (though the syntax is `"$(basename "$1")"`). I'll fix it.

`basename` doesn't strip the extension unless you tell it to, though. So that parts fine.